lgc/util/llpcInternal.cpp

```diff
@@ -81,6 +82,18 @@ CallInst* EmitCall(
     return builder.CreateNamedCall(funcName, pRetTy, args, attribs);
 }
 
+// =====================================================================================================================
+// Emits a amdgcn.reloc.constant intrinsics that represents a relocatable value with the given symbol name.
+llvm::CallInst* EmitRelocationConstant(
+    llvm::IRBuilderBase* pBuilder,             // [in,out] An IRBuilder for instruction insertion
+    const llvm::Twine& symbolName)             // Name of the relocation symbol associated with this relocation
+{
+    auto pMDNode = llvm::MDNode::get(pBuilder->getContext(), llvm::MDString::get(pBuilder->getContext(),
+                                     symbolName.str()));
+    return pBuilder->CreateIntrinsic(Intrinsic::amdgcn_reloc_constant, {},
```

Comment:
Does it work to have an external global variable (perhaps in the const32 address space so it has a 32-bit address) and use the address of that as the constant? Then we wouldn't need the special amdgcn_reloc_constant intrinsic.

Reply:
To answer my own question:

No that does not work. It tries to get an address to the external global variable in a PC-relative way, which is not what we want.